akka-remote/src/main/scala/akka/remote/serialization/DaemonMsgCreateSerializer.scala

```diff
@@ -129,13 +132,44 @@ private[akka] class DaemonMsgCreateSerializer(val system: ExtendedActorSystem) e
       supervisor = deserializeActorRef(system, proto.getSupervisor))
   }
 
-  protected def serialize(any: Any): ByteString = ByteString.copyFrom(serialization.serialize(any.asInstanceOf[AnyRef]).get)
+  private def oldSerialize(any: Any): ByteString = ByteString.copyFrom(serialization.serialize(any.asInstanceOf[AnyRef]).get)
+
+  private def serialize(any: Any): (Int, Boolean, String, Array[Byte]) = {
+    val m = any.asInstanceOf[AnyRef]
+    val serializer = serialization.findSerializerFor(m)
+
+    // this trixery is to retain backwards wire compatibility while at the same time
+    // allowing for usage of string manifests when enable-additional-serialization-bindings = on
+    var hasManifest = false
+    val manifest = serializer match {
+      case ser: SerializerWithStringManifest ⇒
+        hasManifest = true
+        ser.manifest(m)
+      case _ if m eq null ⇒ "null"
+      case _ ⇒
+        val className = m.getClass.getName
```

Comment:
but if it's a real manifest I think we should

    if (ser.includeManifest)
      hasManifest = true

and using the classname as the manifest is the right thing